flake8_trio/visitors/helpers.py

```diff
@@ -181,20 +167,23 @@ def iter_guaranteed_once_cst(iterable: cst.BaseExpression) -> bool:
             func=m.Name("range"),
         ),
     ):
-        try:
-            return (
-                len(
-                    range(
-                        *[
-                            cst_literal_eval(a.value)
-                            for a in ensure_type(iterable, cst.Call).args
-                        ]
-                    )
-                )
-                > 0
-            )
-        except Exception:  # noqa: PIE786
+        values: list[int] = []
+        for arg_arg in ensure_type(iterable, cst.Call).args:
+            arg = arg_arg.value
+            if isinstance(arg, cst.UnaryOperation):
+                if not isinstance(arg.expression, cst.Integer):
+                    return False
+                value = arg.expression.evaluated_value
+                if isinstance(arg.operator, cst.Minus):
+                    value = -value
+            elif isinstance(arg, cst.Integer):
+                value = arg.evaluated_value
+            else:
+                return False
+            values.append(value)
+        if len(values) not in (1, 2, 3):
             return False
+        return len(range(*values)) > 0
```

Comment:
Raises an error a lint time if it would at runtime!  In which case I think we want to return `False`, in case the exception is caught without a checkpoint occuring.

Reply:
Bringing it out to #147